docs/vaahflutter/core/api.md

```diff
@@ -0,0 +1,241 @@
+# API 
+
+::: warning Dependencies
+
+- [dio: ^4.0.6](https://pub.dev/packages/dio)
+
+- This panel uses EnvironmentConfig, thus depending on the env.dart file. It uses the base API URL from your EnviromentConfig.
+:::
+
+[[toc]]
+
+## Change Base URL
+
+To change the base URL change the variable in your [EnvironmentConfig](./environments).
+```dart
+'default': defaultConfig.copyWith(
+    apiUrl: 'YOUR_API_URL',
+),
+```
+The developer should not include the `/` at the end of the URL. e.g.
+```dart
+'default': defaultConfig.copyWith(
+    apiUrl: 'https://api.webreinvent.com',
+),
+```
+
+depending on `enableApiLogs` from [EnviromentConfig](./environments), `LogInterceptor` for dio can be enabled/ disabled.
+
+## Initialize Api class
+
+To use interceptors, Base Api Url, Cookie jar, etc, Api needs to be initialized first before use, so we initialize it using `Api.init()` in [BaseController](../app.md/#base-controller)
+
+```dart
+Api.init();
+```
+
+## Ajax method
+
+- ajax method automatically converts objects from `snake_case To lowerCamelCase` when `Receiving the response`.
```

Comment:
@we-prajapati-c001 keep first letter as Capital

Reply:
I've updated this and pushed a new commit.